moderation.go

```diff
@@ -74,6 +135,24 @@ type ResultCategoryScores struct {
 	SexualMinors          float32 `json:"sexual/minors"`
 	Violence              float32 `json:"violence"`
 	ViolenceGraphic       float32 `json:"violence/graphic"`
+	Illicit               float32 `json:"illicit"`
+	IllicitViolent        float32 `json:"illicit/violent"`
+}
+
+type CategoryAppliedInputType struct {
```

Comment:
@chenhhA thank you for the PR! In the OpenAI API reference, items here might contain `text` or `image` values, while `ModerationItemType` contains `text` and `image_url`.

Could you please add an integration test as well, it is challenging to nail down all the details!

Doc link: https://platform.openai.com/docs/api-reference/moderations/create?lang=curl

<img width="1182" alt="image" src="https://github.com/user-attachments/assets/1b28498d-0672-4b28-b54e-68590f85e1b9">